lightning/src/util/sweep.rs

```diff
@@ -588,6 +601,8 @@ where
 			}
 			true
 		});
+
+		sweeper_state.dirty = true;
```

Comment:
nit: since this changes existing behavior, in the future I'd prefer if this went in a separate commit with an explanation

Reply:
Do you mean setting the dirty flag at a lower level than the original persist was located? Open to splitting the commit, but how?

Reply:
Oh, I thought this line added an extra persist that wasn't present before. If that's wrong, please disregard.